### ISSUE TITLE
head_branch may not be required when checks.create and checks.update

### ISSUE THREAD

**Issue by ylemkimon** [octokit/octokit.js#874]
Although [API docs](https://developer.github.com/v3/checks/runs/#create-a-check-run) states `head_branch` is required when creating or updating a check run, when creating or updating one for commits from forked repository via PR, it isn't, and should not be, required. 

It seems that GitHub currently accepts it, and Travis CI is using it.

**Comment by gr2m** [comment]
Could you provide an example code that shows it’s not required? That would help me file Anbisse with the docs team. All their fixes make it into this Library within a day, it’s mostly automated

**Comment by ylemkimon** [comment]
I'm developing a bot and
**(i)** a pull request is created or synchronized
**(ii)** it receives `pull_request.created|synchronize` event via webhook and requests a check suite
```js
        octokit.request({
            method: 'POST',
            // https://github.com/octokit/rest.js/issues/869
            url: '/repos/:owner/:repo/check-suite-requests',
            headers: {
                accept: 'application/vnd.github.antiope-preview+json',
            },
            owner,
            repo,
            head_sha,
        });
```

**(iii)** GitHub sends a `check_suite.requested` event via webhook with payload, where `check_suite.head_branch` is null
**(iv)** the bot creates a check run without head_branch:
```js
        octokit.request({
            method: 'POST',
            // https://github.com/octokit/rest.js/issues/862
            url: '/repos/:owner/:repo/check-runs',
            headers: {
                accept: 'application/vnd.github.antiope-preview+json',
            },
            name,
            head_sha,
            status: 'in_progress',
            output: {
                title,
                summary,
            },
        });
```

**(v)** a check run for the commit is successfully created.

**Comment by ylemkimon** [comment]
@gr2m I'm developing a bot and
**(i)** a pull request from forked repository is opened or synchronized
**(ii)** it receives [`pull_request.opened|synchronize`](https://developer.github.com/v3/activity/events/types/#pullrequestevent) event via webhook and [requests a check suite](https://developer.github.com/v3/checks/suites/#request-check-suites)
```js
octokit.request({
    method: 'POST',
    // https://github.com/octokit/rest.js/issues/869
    url: '/repos/:owner/:repo/check-suite-requests',
    headers: {
        accept: 'application/vnd.github.antiope-preview+json',
    },
    owner,
    repo,
    head_sha,
});
```

**(iii)** GitHub sends a [`check_suite.requested`](https://developer.github.com/v3/activity/events/types/#checksuiteevent) event via webhook with payload, where `check_suite.head_branch` is `null`
**(iv)** the bot requests to [create a check run](https://developer.github.com/v3/checks/runs/#create-a-check-run) without `head_branch`:
```js
octokit.request({
    method: 'POST',
    // https://github.com/octokit/rest.js/issues/862
    url: '/repos/:owner/:repo/check-runs',
    headers: {
        accept: 'application/vnd.github.antiope-preview+json',
    },
    owner,
    repo,
    name,
    head_sha,
    status: 'in_progress',
    output: {
        title,
        summary,
    },
});
```

**(v)** a check run for the commit is successfully created
**(vi)** same for [updating a check run](https://developer.github.com/v3/checks/runs/#update-a-check-run).

**Comment by ylemkimon** [comment]
It would also be nice to update [Event docs](https://developer.github.com/v3/activity/events/types/#checksuiteevent) that `head_branch` of the payload can be null. ((iii) of https://github.com/octokit/rest.js/issues/874#issuecomment-389043307)

**Comment by ylemkimon** [comment]
@gr2m Do you have updates on the issue?

**Comment by ylemkimon** [comment]
@gr2m Yes, but it's not urgent. Thank you!

**Comment by gr2m** [comment]
Quick update: looks like the `head_branch` parameter is now ignored altogether. Once it’s removed from the docs I’ll remove it from the route specifications, too

**Comment by ylemkimon** [comment]
@gr2m I'd like to know there is any development on the updating the documentation.

**Comment by gr2m** [comment]
The documentation team has it on their roadmap, but didn’t get to it yet.

If this is blocking you I can add a workaround until they do

**Comment by gr2m** [comment]
Or as a workaround on your side you can add anything you want for `head_branch` as it looks like it’s  being ignored on the server anyway

**Comment by ylemkimon** [comment]
@gr2m Passing empty string seems to work. Thank you!

**Comment by bsiegel** [comment]
FYI, ran into this today too and the docs appear to be updated on GitHub's site: https://developer.github.com/v3/checks/runs/#create-a-check-run

**Comment by gr2m** [comment]
Yes, routes got updated here: https://github.com/octokit/routes/pull/198#pullrequestreview-135880990, but I just realised I didn’t squash the commits to use the correct commit message so no release was created. I just merged another PR on octokit/routes which now will trigger a release. Once the PR comes in form Greenkeeeper this issue should be resolved

**Comment by gr2m** [comment]
fixed via #957
